query-algebrizer/src/clauses/predicate.rs

```diff
@@ -134,14 +134,21 @@ impl ConjoiningClauses {
         // We expect the intersection to be Long, Long+Double, Double, or Instant.
         let left_v;
         let right_v;
+        if shared_types.is_empty() {
```

Comment:
We already check `shared_types.is_empty()` in line 115. The result of that check, if the set is empty, is to `mark_known_empty` and return `Ok(())`. As we do nothing else between that check and this, this code will _never_ succeed. Do we really need it?

Reply:
Thanks! Probably a screwup when rebasing a very old branch.